code/processes/wdb.q

```diff
@@ -313,7 +313,15 @@ merge:{[dir;pt;tableinfo;mergelimits;hdbsettings]
 	      curr[0]:curr[0],select from get segment;
 	      curr[1]:curr[1],segment;		
       	$[islast or mergemaxrows < count curr[0];
-	        [.lg.o[`merge;"upserting ",(string count curr[0])," rows to ",string dest];
+	        
+                [pattrtest:@[{@[x;y;`p#]}[curr[0];];`sym;{x}];
```

Comment:
I think it would be better if this test returned a true or false rather than a list or a string
Also, you should have log lines around this test condition as it will not take 0 time - so it would be good to know how long it takes. 
And you have `sym hardcoded in here, but I think it should be .merge.getextrapartitiontype[tablename]

Reply:
For the log lines around the test, do you just mean something like "Starting test" and "Test completed" for example?

Reply:
Yes. You can probably get away with just the "starting test" line as it will be closely followed by either "the p attribute can now be applied" or "upserting.... "